raxx_elli/lib/raxx/elli/handler.ex

```diff
@@ -30,10 +30,19 @@ defmodule Raxx.Elli.Handler do
     headers = :elli_request.headers(elli_request) |> Enum.map(fn ({k, v}) ->
       {String.downcase(k), String.downcase(v)}
     end)
-    {"Host", authority} = :elli_request.headers(elli_request) |> List.keyfind("Host", 0)
-    [host, port] = String.split(authority, ":")
+
+	[host, port] = 
+	case :elli_request.headers(elli_request) |> List.keyfind("Host", 0) do
+	  {"Host", authority} ->
+		case String.split(authority, ":") do
+			[host, port] -> [host, port]
+			[host] -> [host, "80"] # we need to set the correct port some how...
```

Comment:
Regarding `scheme`, `host `and `port `we should be able to get it all directly from elli if we can get them to stop discarding it: (see line #L689)

https://github.com/elli-lib/elli/blob/develop/src/elli_http.erl#L683-L691

```erl
parse_path({abs_path, FullPath}) ->
    case binary:split(FullPath, [<<"?">>]) of
        [URL]       -> {ok, {FullPath, split_path(URL), []}};
        [URL, Args] -> {ok, {FullPath, split_path(URL), split_args(Args)}}
    end;
parse_path({absoluteURI, _Scheme, _Host, _Port, Path}) -> # why is elli discarding scheme, host and port
    parse_path({abs_path, Path});
parse_path(_) ->
    {error, unsupported_uri}.
```